test/end-to-end/oauth2provider.test.js

```diff
@@ -0,0 +1,187 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Imports
+ */
+
+import assert from "assert";
+import puppeteer from "puppeteer";
+import {
+    clearBrowserCookiesWithoutAffectingConsole,
+    toggleSignInSignUp,
+    screenshotOnFailure,
+    backendBeforeEach,
+    waitForUrl,
+    createOAuth2Client,
+    setOAuth2ClientIdInStorage,
+    removeOAuth2ClientIdFromStorage,
+    getOAuth2LoginButton,
+    getOAuth2LogoutButton,
+    getOAuth2TokenData,
+    isReact16,
+    waitFor,
+    signUp,
+    getDefaultSignUpFieldValues,
+    getTestEmail,
+} from "../helpers";
+import fetch from "isomorphic-fetch";
+
+import { TEST_CLIENT_BASE_URL, TEST_SERVER_BASE_URL, SIGN_OUT_API } from "../constants";
+
+/*
+ * Tests.
+ */
+describe("SuperTokens OAuth2Provider", function () {
+    let browser;
+    let page;
+    let consoleLogs = [];
+
+    before(async function () {
+        // Skip these tests if running in React 16
+        if (isReact16()) {
+            this.skip();
+        }
+
+        await backendBeforeEach();
+
+        await fetch(`${TEST_SERVER_BASE_URL}/startst`, {
+            method: "POST",
+        }).catch(console.error);
+
+        browser = await puppeteer.launch({
+            args: ["--no-sandbox", "--disable-setuid-sandbox"],
+            headless: true,
+        });
+    });
+
+    after(async function () {
+        await browser.close();
```

Comment:
Please check that this "passes" (is properly skipped) when running with react16: I have a suspicion the after fn runs anyway and then `browser.close` fails.

Reply:
You were right. Used a flag to prevent running `after` block if it was skipped.